src/main/java/org/owasp/dsomm/metricCA/analyzer/yamlDeserialization/Application.java

```diff
@@ -1,54 +1,59 @@
 package org.owasp.dsomm.metricCA.analyzer.yamlDeserialization;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import org.owasp.dsomm.metricCA.analyzer.yamlDeserialization.components.DatePeriodComponent;
+import org.owasp.dsomm.metricCA.analyzer.yamlDeserialization.components.DatePeriodEndComponent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
 import java.util.Map.Entry;
-import java.util.logging.Logger;
 
 public class Application {
-    private static final Logger LOGGER = Logger.getLogger( Application.class.getName() );
+    private static final Logger logger = LoggerFactory.getLogger(Application.class);
 
+    private String team;
+    private String applicationId;
     private ActivityDirector activityDirector;
 
     public Application(Map<?, ?> configJavaYaml) {
         activityDirector = new ActivityDirector();
         activityDirector.createActivities(configJavaYaml);
     }
 
-    public void saveData(Map<?, ?> appJavaYaml){
+    public void saveData(Map<?, ?> appJavaYaml) {
         HashMap allActivities = (HashMap) (appJavaYaml.get("activities"));
-        for (Object activityKey : allActivities.keySet()){
+        for (Object activityKey : allActivities.keySet()) {
             ArrayList data = (ArrayList) allActivities.get(activityKey);
-            for (int i = 0; i<data.size(); i++){
-                activityDirector.getActivities().get(activityKey).addContent();
+            for (int i = 0; i < data.size(); i++) {
+                activityDirector.getActivities().get(activityKey).addContentSkeleton();
             }
             fillActivityContent(data, activityKey);
         }
+        enrichComponentsOfActivities();
     }
 
-    private void fillActivityContent(ArrayList data, Object activity){
-        for (int i = 0; i<data.size(); i++){ // Data in each activity
+    private void fillActivityContent(ArrayList data, Object activity) {
+        for (int i = 0; i < data.size(); i++) { // Data in each activity
             HashMap<String, Object> temp_data = (HashMap<String, Object>) data.get(i);
             Set<Entry<String, Object>> entrySet = temp_data.entrySet();
+            HashMap comp = temp_data;
+            Iterator<Object> iterator = comp.keySet().iterator();
```

Comment:
@sumsumcity please check this fix of iterators. Beforehand, it didn't iterate and that is why there was no date on TMs